examples/pigweed-app/nrfconnect/README.md

```diff
@@ -0,0 +1,232 @@
+# CHIP nRF Connect nRF52840 Pigweed Example Application
+
+An example application showing the use
+[CHIP](https://github.com/project-chip/connectedhomeip) on the Nordic nRF52840.
+
+<hr>
+
+-   [CHIP nRF52840 Pigweed Example Application](#chip-nrf52840-pigweed-example-application)
+    -   [Introduction](#introduction)
+    -   [Building](#building)
+        -   [Using Docker container](#using-docker-container)
+        -   [Using Native shell](#using-native-shell)
+        -   [Supported nRF Connect SDK versions](#supported-nrf-connect-sdk-versions)
+    -   [Configuring the example](#configuring-the-example)
+    -   [Flashing and debugging](#flashing-and-debugging)
+    -   [Currently implemented features](#currently-implemented-features)
+
+<hr>
+
+<a name="intro"></a>
+
+## Introduction
+
+![nrf52840 DK](../../platform/nrf528xx/doc/images/nrf52840-dk.jpg)
+
+The nRF52840 Pigweed example app exercises functionalities in
+third_party/pigweed, to see what is needed for integrating Pigweed to CHIP, as
+well as a precursor to CHIP functionalities like on-device testing.
+
+The example makes use of the CMake build system to generate the ninja build
+script. The build system takes care of invoking the CHIP library build with all
+necessary flags exported from the Zephyr environment.
+
+<a name="building"></a>
+
+## Building
+
+### Using Docker container
+
+> **Important**:
+>
+> Due to
+> [certain limitations of Docker for MacOS](https://docs.docker.com/docker-for-mac/faqs/#can-i-pass-through-a-usb-device-to-a-container)
+> it is impossible to use the Docker container to communicate with a USB device
+> such as nRF 52840 DK. Therefore, MacOS users are advised to follow the
+> [Using Native shell](#using-native-shell) instruction.
+
+The easiest way to get started with the example is to use nRF Connect SDK Docker
+image for CHIP applications. Run the following commands to start a Docker
+container:
+
+        $ mkdir ~/nrfconnect
+        $ mkdir ~/connectedhomeip
+        $ docker pull nordicsemi/nrfconnect-chip
+        $ docker run --rm -it -e RUNAS=$(id -u) -v ~/nrfconnect:/var/ncs -v ~/connectedhomeip:/var/chip \
+            -v /dev/bus/usb:/dev/bus/usb --device-cgroup-rule "c 189:* rmw" nordicsemi/nrfconnect-chip
```

Comment:
@Damian-Nordic could you comment on this, as I have never met with project-chip/chip-build-nrf-platform image?

Reply:
Sure. We introduced `nordicsemi/nrfconnect-chip` for several reasons:
1) `chip-build-nrf-platform` comes with embedded nRF Connect sources to which you don't have access outside the container while `nordicsemi/nrfconnect-chip` fetches SDK to a mounted host directory. I find it quite useful to be able to configure IDE to see headers from the SDK.
2) `chip-build-nrf-platform` doesn't support building as non-root user
3) `nordicsemi/nrfconnect-chip` shortens development time for us as we don't need to wait for a) PR with Dockerfile changes to be merged b) the new version to be pushed to the docker hub c) do dependent changes in README. We can do a) & b) instantly and focus on c).
4) `nordicsemi/nrfconnect-chip` is much smaller in size and faster to setup. 

I admit 1) and 2) can be addressed by adding support for nRF Connect SDK to the VS Code container (which will be added by the end of this week), so we may rethink using `nordicsemi/nrfconnect-chip` once that is done although the VS Code container is quite large, time-consuming to initialize and may seem like a bit overkill for someone willing to try a single example. 3) and 4) were important for us, especially in the initial phase of CHIP development, but if you think CHIP shouldn't rely on external images, we won't insist to use it.

Reply:
It seems to me that the discussion turned to more general topics, which are not strictly related to this PR, so what do you think about creating some issue and continuing it there, so this one could not be blocked?

Reply:
> Sure. We introduced `nordicsemi/nrfconnect-chip` for several reasons:
> 
> 1. `chip-build-nrf-platform` comes with embedded nRF Connect sources to which you don't have access outside the container while `nordicsemi/nrfconnect-chip` fetches SDK to a mounted host directory. I find it quite useful to be able to configure IDE to see headers from the SDK.

This is fixable, and should be fixed. We shouldn't have to rebuild docker images to bump the revision.

> 2. `chip-build-nrf-platform` doesn't support building as non-root user

Also fixable, and should be fixed.

> 3. `nordicsemi/nrfconnect-chip` shortens development time for us as we don't need to wait for a) PR with Dockerfile changes to be merged b) the new version to be pushed to the docker hub c) do dependent changes in README. We can do a) & b) instantly and focus on c).

This problem seems symmetric. Now we have the same problem, but instead CHIP's documented flows blocks on Nordic updating its containers. I think CHIP should own the images, so that our CI & documentation stay in sync.

I do however agree our current docker process needs improvement; ideally GitHub automation would handle building and pushing images from dockerfile changes. That's the way forward in my opinion.

> 4. `nordicsemi/nrfconnect-chip` is much smaller in size and faster to setup.

Is this just a result of 1) ?

> 
> I admit 1) and 2) can be addressed by adding support for nRF Connect SDK to the VS Code container (which will be added by the end of this week), so we may rethink using `nordicsemi/nrfconnect-chip` once that is done although the VS Code container is quite large, time-consuming to initialize and may seem like a bit overkill for someone willing to try a single example. 3) and 4) were important for us, especially in the initial phase of CHIP development, but if you think CHIP shouldn't rely on external images, we won't insist to use it.

I see the argument that your image works better right now, but I'd ask you to contribute those improvements.

Reply:
> It seems to me that the discussion turned to more general topics, which are not strictly related to this PR, so what do you think about creating some issue and continuing it there, so this one could not be blocked?

Sounds good.

Reply:
https://github.com/project-chip/connectedhomeip/issues/3687